src/test/java/nextstep/courses/domain/PaidRegistrationPolicyTest.java

```diff
@@ -0,0 +1,54 @@
+package nextstep.courses.domain;
+
+import java.time.LocalDateTime;
+
+import org.junit.jupiter.api.Test;
+
+import nextstep.users.domain.NsUserTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchIllegalArgumentException;
+import static org.junit.jupiter.api.Assertions.*;
+
+class PaidRegistrationPolicyTest {
+
+    @Test
+    void 유료_강의는_강의_최대_수강_인원을_초과할_수_없다() {
+        int maxStudentCount = 1;
+        int sessionFee = 20000;
+
+        CoverImage coverImage = CoverImageTest.createCoverImage1();
+        LocalDateTime startedAt = LocalDateTime.of(2023, 10, 1, 0, 0, 0);
+        LocalDateTime endedAt = LocalDateTime.of(2023, 10, 1, 23, 0, 0);
+
+        Session session
+            = SessionFactory.ofPaid(1, coverImage, SessionStatus.RECRUITING, sessionFee, maxStudentCount, startedAt, endedAt);
+
+        IllegalArgumentException e = catchIllegalArgumentException(() -> {
+            session.register(NsUserTest.JAVAJIGI, new Money(sessionFee));
+            session.register(NsUserTest.SANJIGI, new Money(sessionFee));
+        });
+
+        assertThat(e).hasMessage("강의 최대 수강 인원을 초과할 수 없습니다.");
+    }
```

Comment:
이 테스트에서 필요한 given 절은 무엇일까요?
유료여부, 최대수강인원 2가지만 given 절에 존재해야합니다!
지금은 이미지, 수강기간과 같은 불필요한 값들이 존재하군요 🤔

이렇게 불필요한 값들이 존재하면 테스트의 의도가 명확하지 않을 수 있고 팀원들이 봤을때를 생각해서 불필요한 값이 있다면 코드 읽는게 오래걸려 피곤해지기도하고 다른 팀원들의 시간도 소중하니 중요한 부분이라고 생각해요.

테스트 코드도 결국 우리가 함께 읽고 유지보수하는 소중한 자산이니, 핵심만 담아 더 명확하게 작성하면 어떨까요? 😊

```suggestion
    @Test
    void 유료_강의는_강의_최대_수강_인원을_초과할_수_없다() {
        // Given - 필요한 값만 명시적으로 설정
        Session session = SessionBuilder.aPaidSession()
                .withMaxStudentCount(1)
                .withFee(20000)
                .build();

        // When & Then
        IllegalArgumentException e = catchIllegalArgumentException(() -> {
            session.register(NsUserTest.JAVAJIGI, new Money(20000));
            session.register(NsUserTest.SANJIGI, new Money(20000));
        });
        assertThat(e).hasMessage("강의 최대 수강 인원을 초과할 수 없습니다.");
    }
```

이렇게 테스트픽스쳐를 만들어서 테스트에 필요한 부분에만 집중하고 검증할 수 있게 리팩토링해보는건 어떨까요? 🤔
도움이 될만한 [글](https://www.arhohuttunen.com/test-data-builders/) 남겨드릴게요
꼭 읽어보시고 다른 테스트코드도 각각 필요한 기능에 초점을 맞출 수 있게 작성 부탁드립니다 🙇

Reply:
@mskangg 
좋은 리뷰 주셔서 감사합니다.

적절한 디폴트값을 주어서 처리하는 방법을 제안주신것 같네요.
이부분에 대해 고민이 많았는데 역시나 빌더 패턴이 필요하겠군요. 귀차니즘..
수정해보겠습니다.